samza-core/src/main/java/org/apache/samza/zk/ZkLeaderElector.java

```diff
@@ -105,7 +91,7 @@ public boolean tryBecomeLeader() {
     if (index == 0) {
       isLeader.getAndSet(true);
       LOGGER.info(zLog("Eligible to become the leader!"));
-      zkLeaderElectorListener.onBecomingLeader(); // inform the caller
+      leaderElectorListener.onBecomingLeader(); // inform the caller
```

Comment:
With your example, the "ClassThatNeedsLeaderElector" should be instantiating the instance of ZkLeaderElector right? Anyway.. I think I am not getting the whole picture of the use-case you propose. I just thought it might simplify implementation. If you think it makes the class more flexible to use, go for it :) 

Going by the same semantics, wouldn't also need a callback for the follower? :)